controllers/resource.go

```diff
@@ -60,6 +60,14 @@ func (r *OperatorPipelineReconciler) reconcileResources(ctx context.Context, pip
 		return err
 	}
 
+	if err := r.ensureDockerRegistrySecret(ctx, pipeline.ObjectMeta); err != nil {
+		log.Info("Docker Registry Secret not present or correct. Create it to use a private repository.")
```

Comment:
IMO, using log.Warning might be more appropriate since we are handling an error.

Reply:
hmm....looks like our logger doesn't have warning? weird. looking into it

Reply:
yeah, okay. no such thing as warning or debug in this logger. it has V-levels
https://pkg.go.dev/github.com/go-logr/logr#section-readme

I could put like `log.V(2).Info("Docker Registry secret message")`
but I wouldn't say that's necessarily more clear and it would disappear if verbosity is turned down

Reply:
@dcurran90 is correct, there is no `warn` in the library used by controller runtime. I'm fine with this being an `info` and I think it would be bad user experience to log `error` for something that's optional.

Reply:
We should still propagate the err itself, unless it's a `errors.IsNotFound(err)`.

Reply:
@itroyano has a good point.